tests/Shared/WorkloadTesting/TemplateCustomHive.cs

```diff
@@ -90,7 +90,7 @@ await InstallTemplatesAsync(
 
     public static string GetPackagePath(string builtNuGetsPath, string templatePackageId)
     {
-        var packageNameRegex = new Regex($@"{templatePackageId}\.\d+\.\d+\.\d+(-[A-z]*\.*\d*)?\.nupkg");
+        var packageNameRegex = new Regex($@"{templatePackageId}\.\d+\.\d+\.\d+(-[A-z\.\d]*\.*\d*)?\.nupkg");
```

Comment:
It's necessary to encode the package id or some char in it could have a meaning (`.`, `-`). Then instead of discovering new ways the version can be encoded, I would suggest to accept anything until the `.nupkg` extension.

```suggestion
        var packageNameRegex = new Regex($@"{Regex.Escape(templatePackageId)}.*\.nupkg");
```

Reply:
I'll fix this in a follow up. An alternative would be to find all files with `{packageName}.*nupkg`, and that the "rest" of the string" starts with `.\d`? But that would fail if we had a package with a number in it then.